pilot/pkg/model/validation.go

```diff
@@ -1911,27 +1911,34 @@ func ValidateEndUserAuthenticationPolicySpecBinding(msg proto.Message) error {
 
 // ValidateVirtualService checks that a v1alpha3 route rule is well-formed.
 func ValidateVirtualService(msg proto.Message) (errs error) {
-	routeRule, ok := msg.(*networking.VirtualService)
+	virtualService, ok := msg.(*networking.VirtualService)
 	if !ok {
-		return errors.New("cannot cast to v1alpha3 routing rule")
+		return errors.New("cannot cast to virtual service")
 	}
 
-	// TODO: routeRule.Gateways
+	for _, gateway := range virtualService.Gateways {
+		if !IsDNS1123Label(gateway) { // FIXME: verify this restriction
```

Comment:
@rshriram @ZackButcher ^^

Reply:
More and more I'm convinced we need a distinction between names and hostnames; but for now since we use CRDs gateway names must also be valid DNS names so this is fine.